astropy/io/fits/tests/test_core.py

```diff
@@ -593,6 +595,48 @@ def test_open_nonexistent(self):
             assert os.path.exists(self.temp('foobar.fits'))
             os.remove(self.temp('foobar.fits'))
 
+    @pytest.mark.skipif(six.PY2,
```

Comment:
Oddly enough, it looks like you still can't use `urllib` from `six.moves` exactly the same way in Python 2.7 and 3. For example, in Python 3 I can use the result of `urllib.request.urlopen` in a `with` context, but I can't do that in Python 2.7. I can work around this with some version-specific code in the unit tests if you want, but I'm not sure it's worth it.